### PR TITLE
New topic: Plugin Alternatives

### PR DESCRIPTION
**Pull request by karollewandowski**

### CODE REVIEW

File: topics/basics/plugin_alternatives.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+In some cases, implementing an actual IntelliJ Platform plugin can be overkill, and using one of the alternative approaches listed below may provide you with the required value in a much shorter time.
+If you need a functionality that is specific to your project domain, conventions, or practices, you can avoid all the steps that are required to implement and publish a plugin and provide these features as a part of your project or IDE configuration files.
+
+Before you start the IntelliJ Platform plugin development, define your requirements and verify if they can be covered with the alternatives described below.
+Consider implementing an actual plugin only when the described solutions are insufficient in your case and there is a significant number of developers who can benefit from it.
+
+## Structural Search and Replace Inspections
+
+The [structural search and replace (SSR)](https://www.jetbrains.com/help/idea/structural-search-and-replace.html) functionality allows defining search patterns which are based not only on textual information but also on the structure of the searched code fragments, no matter how it is formatted or commented.
```
Comment (YannCebron) [inline topics/basics/plugin_alternatives.md:13]: Structural Search and Replace

File: topics/basics/plugin_alternatives.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Created scripts are stored in the [IDE configuration directory](https://www.jetbrains.com/help/idea/directories-used-by-the-ide-to-store-settings-caches-plugins-and-logs.html#config-directory) and can't be shared as part of project files or configuration.
+
+## Flora Plugin
+
+The [Flora](https://plugins.jetbrains.com/plugin/17669-flora-beta-) plugin allows for developing project-specific extensions as Kotlin Script (<path>\*.kts</path>) or JavaScript (<path>\*.js</path>) files.
+Flora extensions have access to all available IntelliJ Platform APIs, just like a regular plugin.
+
+Every extension is represented by a single file and stored directly in a project's <path>.plugins</path> directory.
+Extensions can be easily shared with other team members by adding the <path>.plugins</path> directory to VCS.
+Also, adding the Flora plugin in the <menupath>Build, Execution, Deployment | Required Plugins</menupath> and sharing this configuration as part of a project makes it effortless to deliver additional IDE functionalities to your team without any manual setup.
```
Comment (YannCebron) [inline topics/basics/plugin_alternatives.md:39]: Settings/Preferences | Build, ....

File: topics/basics/plugin_alternatives.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+The [LivePlugin](https://plugins.jetbrains.com/plugin/7282-liveplugin) allows for extending IntelliJ-based IDEs functionalities the runtime, without the need of restarting IDE.
+It adds a new <control>Plugins</control> tool window that lists all available extensions and allows managing them.
+Extensions can be implemented in Kotlin or Groovy and edited directly in the IDE.
+Extensions can use all IntelliJ Platform APIs and additional LivePlugin API that shorten common use cases.
+
+Created extensions are stored on the IDE level and can be shared with other team members as plain files, GitHub gists, or repositories.
+Additionally, if they are stored in a project's <path>.live-plugins</path> directory and LivePlugin's <control>Run Project Specific Plugins</control> option is enabled, all extensions from this directory will be loaded automatically when the project is opened and unloaded when the project is closed.
+
+> See [LivePlugin author's article](https://dmitrykandalov.com/liveplugin) and [presentation](https://www.youtube.com/watch?v=GcYa4lMRta0) for more information.
```
Comment (YannCebron) [inline topics/basics/plugin_alternatives.md:51]: eventually use `<video>` here for YT link?

Comment (karollewandowski) [reply topics/basics/plugin_alternatives.md]: We would have to remove callout, and still doesn't look good IMHO. I would leave it as is for now and potentially change it in the future.

File: topics/basics/plugin_alternatives.md (hunk 1 above)
Comment (karollewandowski) [reply topics/basics/plugin_alternatives.md]: Fixed, thanks.

**Review by YannCebron — APPROVED** [review]

**Pull request merged**